merlin/dag/ops/subgraph.py

```diff
@@ -0,0 +1,148 @@
+
+        Parameters
+        -----------
+        columns: list of str or list of list of str
+            The columns to apply this operator to
+        transformable: Transformable
+            A pandas or cudf dataframe that this operator will work on
+
+        Returns
+        -------
+        DataFrame
+            Returns a transformed dataframe for this operator
+        """
+        executor = LocalExecutor()
+        transformable = executor.transform(transformable, self.graph)
+        while self.loop_until and not self.loop_until(transformable):
+            transformable = executor.transform(transformable, self.graph)
+        return transformable
+
+    def fit(self, col_selector: ColumnSelector, dataset: Dataset):
+        DaskExecutor().fit(dataset, self.graph)
+
+    def compute_input_schema(
+        self,
+        root_schema: Schema,
+        parents_schema: Schema,
+        deps_schema: Schema,
+        selector: ColumnSelector,
+    ) -> Schema:
+        """
+        Return the schemas of columns
+
+        Parameters
+        ----------
+        root_schema : Schema
+            Schema of the columns from the input dataset
+        parents_schema : Schema
+            Schema of the columns from the parent nodes
+        deps_schema : Schema
+            Schema of the columns from the dependency nodes
+        selector : ColumnSelector
+            Existing column selector for this node in the graph (often None)
+
+        Returns
+        -------
+        Schema
+            Schema of selected columns from input, parents, and dependencies
+        """
+        if not self.graph.input_schema:
+            input_schema = super().compute_input_schema(
+                root_schema, parents_schema, deps_schema, selector
+            )
+            self.graph = self.graph.construct_schema(input_schema)
+
+        return self.graph.input_schema
+
+    def compute_output_schema(
+        self,
+        input_schema: Schema,
+        col_selector: ColumnSelector,
+        prev_output_schema: Schema = None,
+    ) -> Schema:
+        """Given a set of schemas and a column selector for the input columns,
+        returns a set of schemas for the transformed columns this operator will produce
+
+        Parameters
+        -----------
+        input_schema: Schema
+            The schemas of the columns to apply this operator to
+        col_selector: ColumnSelector
+            The column selector to apply to the input schema
+        Returns
+        -------
+        Schema
+            The schemas of the columns produced by this operator
+        """
+        return self.graph.output_schema
+
+    def clear(self):
+        """Removes calculated statistics from each node in the graph
+
+        See Also
+        --------
+        StatOperator.clear
+        """
+        for stat in Graph.get_nodes_by_op_type([self.graph.output_node], StatOperator):
+            stat.op.clear()
+
+    def column_mapping(self, col_selector):
```

Comment:
It turns out to be crucially important that the `Subgraph` operator properly delegates the job of creating a mapping that connects outputs with inputs 😅